brd-android/trade/src/main/java/com/fabriik/trade/ui/features/swap/SwapInputViewModel.kt

```diff
@@ -640,6 +703,30 @@ class SwapInputViewModel(
         }
     }
 
+    private suspend fun estimateFeeBasis(orderAddress: String, currency: String, orderAmount: BigDecimal) : TransferFeeBasis? {
+        val wallet = breadBox.wallet(currency).first()
+
+        // Skip if address is not valid
+        val address = wallet.addressFor(orderAddress) ?: return null
+        if (wallet.containsAddress(address))
+            return null
+        val amount = Amount.create(orderAmount.toDouble(), wallet.unit)
+        val networkFee = wallet.feeForSpeed(TransferSpeed.Regular(currency))
+
+        return try {
+            val data = wallet.estimateFee(address, amount, networkFee)
+            val fee = data.fee.toBigDecimal()
+            check(!fee.isZero()) { "Estimated fee was zero" }
+            data
+        } catch (e: FeeEstimationError) {
```

Comment:
Could we merge both catch clauses into one?

Reply:
@roknovak please, apply this comment on the next PR. I will merge this now.